tests/integ/test_workflow.py

```diff
@@ -1128,6 +1129,123 @@ def test_two_step_lambda_pipeline_with_output_reference(
             pass
 
 
+def test_one_step_emr_pipeline(sagemaker_session, role, pipeline_name, region_name):
+    instance_count = ParameterInteger(name="InstanceCount", default_value=2)
+
+    emr_step_config = EMRStepConfig(
+        jar="s3:/script-runner/script-runner.jar",
+        args=["--arg_0", "arg_0_value"],
+        main_class="com.my.main",
+        properties=[{"Key": "Foo", "Value": "Foo_value"}, {"Key": "Bar", "Value": "Bar_value"}],
+    )
+
+    step_emr = EMRStep(
+        name="emr-step",
+        cluster_id="MyClusterID",
+        display_name="emr_step",
+        description="MyEMRStepDescription",
+        step_config=emr_step_config,
+    )
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[instance_count],
+        steps=[step_emr],
+        sagemaker_session=sagemaker_session,
+    )
+
+    try:
+        response = pipeline.create(role)
+        create_arn = response["PipelineArn"]
+
+        execution = pipeline.start()
+        response = execution.describe()
+        assert response["PipelineArn"] == create_arn
+
+        try:
+            execution.wait(delay=60, max_attempts=10)
+        except WaiterError:
+            pass
+
+        execution_steps = execution.list_steps()
+        assert len(execution_steps) == 1
+        assert execution_steps[0]["StepName"] == "emr-step"
+    finally:
+        try:
+            pipeline.delete()
+        except Exception:
+            pass
+
+
+def test_two_steps_emr_pipeline_without_nullable_config_fields(
+    sagemaker_session, role, pipeline_name, region_name
+):
+    instance_count = ParameterInteger(name="InstanceCount", default_value=2)
+
+    emr_step_config_1 = EMRStepConfig(
+        jar="s3:/script-runner/script-runner_1.jar",
+        args=["--arg_0", "arg_0_value"],
+        main_class="com.my.main",
+        properties=[{"Key": "Foo", "Value": "Foo_value"}, {"Key": "Bar", "Value": "Bar_value"}],
+    )
+
+    step_emr_1 = EMRStep(
+        name="emr-step-1",
+        cluster_id="MyClusterID",
+        display_name="emr-step-1",
+        description="MyEMRStepDescription",
+        step_config=emr_step_config_1,
+    )
+
+    emr_step_config_2 = EMRStepConfig(jar="s3:/script-runner/script-runner_2.jar")
+
+    step_emr_2 = EMRStep(
+        name="emr-step-2",
+        cluster_id="MyClusterID",
+        display_name="emr-step-2",
+        description="MyEMRStepDescription",
+        step_config=emr_step_config_2,
+    )
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[instance_count],
+        steps=[step_emr_1, step_emr_2],
+        sagemaker_session=sagemaker_session,
+    )
+
+    try:
+        response = pipeline.create(role)
+        create_arn = response["PipelineArn"]
+
+        execution = pipeline.start()
+        response = execution.describe()
+        assert response["PipelineArn"] == create_arn
+
+        try:
+            execution.wait(delay=60, max_attempts=10)
```

Comment:
we have two choices as we discussed: 
1. remove this and only test create/update pipeline
2. keep this but you need to create an actual cluster. 

I think you do have a point; if the pipeline creates, we should be fine. So I prefer the first choice